src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -0,0 +1,451 @@
+  }
+
+  // Lazily, possibly create the database, returning the result in the future
+  private lazy val createDatabase: Future[Unit] = {
+    import _root_.slick.util.ConfigExtensionMethods._
+    if (databaseConfig.getBooleanOr("slick.createSchema")) {
+      database.run(dataAccess.schema.create)
+    } else Future.successful(())
+  }
+
+  // Run action with an outer transaction. Also if we need to, create the in memory database!
+  private def runTransaction[R](action: DBIOAction[R, _ <: NoStream, _ <: Effect]): Future[R] = {
+    for {
+      _ <- createDatabase // If not created
+      result <- database.run(action.transactionally)
+    } yield result
+  }
+
+  /**
+   * Creates a row in each of the backend-info specific tables for each call in `calls` corresponding to the backend
+   * `backend`.  Or perhaps defer this?
+   */
+  override def createWorkflow(workflowInfo: WorkflowInfo,
+                              workflowInputs: Traversable[SymbolStoreEntry],
+                              calls: Traversable[Call],
+                              backend: Backend): Future[Unit] = {
+
+    val action = for {
+
+      workflowExecutionInsert <- dataAccess.workflowExecutionsAutoInc +=
+        new WorkflowExecution(
+          workflowInfo.workflowId.toString,
+          WorkflowSubmitted.toString,
+          new Date().toTimestamp)
+
+      _ <- dataAccess.workflowExecutionAuxesAutoInc += new WorkflowExecutionAux(
+        workflowExecutionInsert.workflowExecutionId.get,
+        workflowInfo.wdlSource,
+        workflowInfo.wdlJson)
+
+      symbolInsert <- dataAccess.symbolsAutoInc ++=
+        workflowInputs.toSeq map { symbol =>
+          new Symbol(
+            workflowExecutionInsert.workflowExecutionId.get,
+            symbol.key.scope,
+            symbol.key.name,
+            symbol.key.iteration,
+            if (symbol.isInput) IoInput else IoOutput,
+            symbol.wdlType.toWdlString,
+            symbol.wdlValue.map(wdlValueToString).orNull)
+        }
+
+      // NOTE: Don't use DBIO.seq for **transforming** sequences
+      // - DBIO.seq(mySeq: _*) runs *any* items in sequence, but converts Seq[ DBIOAction[_] ] to DBIOAction[ Unit ]
+      // - DBIO.sequence(mySeq) converts Seq[ DBIOAction[R] ] to DBIOAction[ Seq[R] ]
+      // - DBIO.fold(mySeq, init) converts Seq[ DBIOAction[R] ] to DBIOAction[R]
+
+      _ <- DBIO.sequence(calls.toSeq map {
+        call => for {
+          executionInsert <- dataAccess.executionsAutoInc +=
+            new Execution(
+              workflowExecutionInsert.workflowExecutionId.get,
+              call.taskFqn,
+              ExecutionStatus.NotStarted.toString)
+
+          _ <- backend match {
+            case _: LocalBackend =>
+              dataAccess.localJobsAutoInc +=
+                new LocalJob(
+                  executionInsert.executionId.get,
+                  -1,
+                  -1)
```

Comment:
hmm maybe these two columns should be nullable rather than using `-1`s as sentinels